cmd/outline-ss-server/main.go

```diff
@@ -58,62 +60,49 @@ func init() {
 	logger = logging.MustGetLogger("")
 }
 
-type ssPort struct {
-	tcpListener *net.TCPListener
-	packetConn  net.PacketConn
-	cipherList  service.CipherList
+type Handler interface {
+	NumCiphers() int
+	AddCipher(entry *service.CipherEntry)
```

Comment:
These are misplaced. AddCipher should be on the service, not the handler. The Handler gets the full list, which is shared.